### PR TITLE
ENH: Lazy tifffile

### PR DESCRIPTION
**Pull request by scottclowe**
Add a new datahandler, DataHandlerTifffileLazy, which uses tifffile to handle TIFFs but does so in a lazy manner, loading the data frame-by-frame when determining the traces (like DataHandlerPillow).

The advantage over DataHandlerPillow is it can handle more dtypes and shapes in the TIFF files.

This replaces DataHandlerPillow as the lowmemory_mode datahandler.

### CODE REVIEW

File: fissa/extraction.py [hunk 1]
```diff
@@ -275,6 +275,169 @@ def extracttraces(data, masks):
...
+    """
+    @staticmethod
+    def image2array(image):
+        """
+        Load a TIFF image from disk.
+
+        Parameters
+        ----------
+        image : str or array_like
+            Either a path to a TIFF file, or array_like data.
```
Comment (swkeemink) [inline fissa/extraction.py:290]: The current function can only take in a TiffFile object, no arrays.

Example:
```
import tifffile
import numpy as np
image = np.random.rand(10, 10,10)
tifffile.TiffFile(image)
```
will give an error. 
(Specifically, TiffFile can take as a first argument 'a file name, seekable binary stream, or FileHandle')

**Review by swkeemink — CHANGES_REQUESTED** [review]
One important potential bug.

We should also clarify in the explanation of the low_memory option in the notebooks that it only applies for loading tiff files which might be too high dimensional, and won't work for the option we provide of just inputting arrays instead of tifffiles (if that's not already done properly). 

(Incidentally perhaps we should have a separate dataloader for arrays, and detect in the beginning if the provided images are tiffs or strings, and then load the appropriate data loader.)

**Comment by scottclowe** [comment]
> One important potential bug.

It's not a bug, it's just the docstring is wrong. Like the Pillow loader, lazy loader *doesn't* support a list of file names.

> We should also clarify in the explanation of the low_memory option in the notebooks that it only applies for loading tiff files which might be too high dimensional, and won't work for the option we provide of just inputting arrays instead of tifffiles (if that's not already done properly).

It isn't only for TIFF files which are high dimensional. It's for any (large) TIFF files.

> (Incidentally perhaps we should have a separate dataloader for arrays, and detect in the beginning if the provided images are tiffs or strings, and then load the appropriate data loader.)

We could. Although you would want to check which dataloader to use for each argument in the list, since there is no guarantee the list is all file name strings or all arrays (it could be a mixture). I don't think this is worth prioritising.

**Comment by swkeemink** [comment]
> We could. Although you would want to check which dataloader to use for each argument in the list, since there is no guarantee the list is all file name strings or all arrays (it could be a mixture). I don't think this is worth prioritising.
Agreed fully, was more of a side-note.

With the docstring update happy with this PR otherwise.

**Comment by scottclowe** [comment]
> With the docstring update happy with this PR otherwise.

Great! Thanks for spotting the inconsistency between docstring and behaviour.

**Pull request merged**